lib/id3c/cli/command/etl/manifest.py

```diff
@@ -204,17 +225,23 @@ def upsert_sample(db: DatabaseSession,
         sample = samples[0]
 
         LOG.info(f"Updating existing sample {sample.id}")
+
+        # Update identifier and collection_identifier if update_identifiers is True
+        identifiers_update = """
+             identifier = %(identifier)s,
+                collection_identifier = %(collection_identifier)s, """  \
+                    if update_identifiers else ""
+
         sample = db.fetch_row("""
             update warehouse.sample
-               set identifier = %(identifier)s,
-                   collection_identifier = %(collection_identifier)s,
-                   collected = coalesce(date_or_null(%(collection_date)s), collected),
-                   details = coalesce(details, '{}') || %(additional_details)s
+                set {identifiers_update}
+                    collected = coalesce(date_or_null(%(collection_date)s), collected),
+                    details = coalesce(details, {empty_json}) || %(additional_details)s
 
              where sample_id = %(sample_id)s
 
             returning sample_id as id, identifier, collection_identifier, encounter_id
-            """,
+            """.format(identifiers_update = identifiers_update, empty_json = "'{}'"),
```

Comment:
This was a good opportunity for me to learn about psycopg2's SQL composition. In this case, it may have been a bit overboard because in here we're not dynamically determining which columns are getting updated .... it's either "include identifier and collection_identifier" or don't. So, the SQL is variable, but not super variable.
I create the first part as its own composable and use it in the main command.